AmplifyPlugins/Auth/AWSAuthPlugin/Support/Utils/AuthUserAttributeKey+Extension.swift

```diff
@@ -43,3 +43,40 @@ extension AuthUserAttributeKey {
         }
     }
 }
+
+extension String {
+
+    func toUserAttributeKey() -> AuthUserAttributeKey {
+        switch self {
+        case AuthUserAttributeKey.address.toString():
+            return .address
+        case AuthUserAttributeKey.birthDate.toString():
+            return .birthDate
+        case AuthUserAttributeKey.email.toString():
+            return .email
+        case AuthUserAttributeKey.familyName.toString():
```

Comment:
are these two trying to convert them back and forth? if the string version of `AuthUserAttributeKey.familyName` is "family_name" then from "family_name" to string will never match the string version of AuthUserAttributeKey.familyName which is "family". 

I don't think we should have an extension on String because that would increase the number of methods possible on all String values, and encapsulate entirely in AuthuserAttributeKey extension, something like
```
struct AWSAttributeKey {
    static let familyName: String = "family_name"
}
extension AuthUserAttributeKey {
   func toString() -> String {
        switch self {
        case .familyName:
            return AWSAttributeKey.familyName
        case .custom(let attribute):
            return "custom: \(attribute)"
        }
    }

   public init(_ value: String) {
        switch value {
        case AWSAttributeKey.familyName:
            self = .familyName
        default:
            self = .custom(value)
        }
    }
}
```